### PR TITLE
Update OpenLineage / Great Expectations integration

### PR DESCRIPTION
**Pull request by jwitz**
Part of https://github.com/astronomer/docs/issues/1575

Tried this setup myself and noticed a few things wrong with our setup: 

- Checkpoint files seem to now be in YAML instead of XML
- We never give guidance on how to structure a GE project within an Astro project
- We need to use the datakin URL for custom checkpoints

### CODE REVIEW

File: astro/set-up-data-lineage.md [hunk 1]
```diff
@@ -179,15 +180,15 @@ If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use
...
               module_name: openlineage.common.provider.great_expectations
-              openlineage_host: https://<your-astro-base-domain>
+              openlineage_host: https://<your-base-domain>.datakin.com
               openlineage_apiKey: <your-lineage-api-key>
               openlineage_namespace: <NAMESPACE_NAME> # Replace with your job namespace; Astronomer recommends using a meaningful namespace such as `dev` or `prod`.
               job_name: validate_my_dataset
     ```
 
-2. Lineage support for GreatExpectations requires the use of the `ActionListValidationOperator`. In each of your checkpoint's xml files in `checkpoints/`, set the `validation_operator_name` configuration to `action_list_operator`:
+2. Lineage support for GreatExpectations requires the use of the `ActionListValidationOperator`. In each of your checkpoint's YAML files in `checkpoints/`, set the `validation_operator_name` configuration to `action_list_operator` like in the following snippet:
```
Comment (kwcanuck) [inline astro/set-up-data-lineage.md:189]:
```suggestion
2. Lineage support for GreatExpectations requires the use of the `ActionListValidationOperator`. In each of your checkpoint's YAML files in `checkpoints/`, set the `validation_operator_name` configuration to `action_list_operator` as shown in the following snippet:
```

**Review by kwcanuck — APPROVED** [review]

File: astro/set-up-data-lineage.md [hunk 2]
```diff
@@ -160,13 +160,14 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
-- A [Great Expectations project](https://legacy.docs.greatexpectations.io/en/latest/guides/tutorials/getting_started.html#tutorials-getting-started).
-- Your Astro base domain.
-- Your Organization's OpenLineage API key.
+- A [Great Expectations Data Context](https://legacy.docs.greatexpectations.io/en/latest/guides/tutorials/getting_started.html#tutorials-getting-started) in the `include` directory of your Astro project.
+- If using a custom Checkpoint or Checkpoint config, your Astro base domain and OpenLineage API key.
 
 #### Setup
 
-If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use a custom Checkpoint or Checkpoint Config, the operator detects your Astro OpenLineage configuration and sends lineage information automatically. If you use custom Checkpoints, complete the following steps:
+If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use a custom Checkpoint or Checkpoint Config, the operator detects your Astro OpenLineage configuration from the `include` folder automatically as long as you set `data_context_root_dir=os.path.join(base_path, "include", "great_expectations"),` as a task-level parameter.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:168]: @denimalpaca I'm having trouble getting my DAG to recognize `base_path`. I tried setting up my DAG similar to [here](https://github.com/astronomer/airflow-data-quality-demo/blob/main/dags/great_expectations/great_expectations_bigquery.py), but the parser isn't finding `usr/local/include/great_expectations/great_expectations.yml` even after adding all of the required imports.

Comment (denimalpaca) [reply astro/set-up-data-lineage.md:168]: Make sure you have the leading slash in `/usr/local/include/great_expectations/great_expectations.yml`, otherwise it's looking for `usr` in your local directory.

Comment (jwitz) [reply astro/set-up-data-lineage.md:168]: Sorry missed that slash in my comment but it exists in the error message. Here's the entire error message:

```

rel_path = 'dags/example_dag_basic.py'
rv = 'Traceback (most recent call last):\n  File "/usr/local/lib/python3.9/site-packages/great_expectations/data_context/da...\n    - If you have never run Great Expectations in this project, please run `great_expectations init` to get started.'

    @pytest.mark.parametrize("rel_path,rv", get_import_errors(), ids=[x[0] for x in get_import_errors()])
    def test_file_imports(rel_path,rv):
    	""" Test for import errors on a file """
    	if rel_path and rv : #Make sure our no op test doesn't raise an error
>   		raise Exception(f"{rel_path} failed to import with message \n {rv}")
E     Exception: dags/example_dag_basic.py failed to import with message 
E      Traceback (most recent call last):
E       File "/usr/local/lib/python3.9/site-packages/great_expectations/data_context/data_context/data_context.py", line 409, in _load_project_config
E         with open(path_to_yml) as data:
E     FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/include/great_expectations/great_expectations.yml'
E     
```

This is after specifying a base path in the DAG. Feel free to take this convo on to Slack if the answer is tricky

Comment (denimalpaca) [reply astro/set-up-data-lineage.md:168]: `E     FileNotFoundError: [Errno 2] No such file or directory: '/usr/local/include/great_expectations/great_expectations.yml'`

path should be `/usr/local/airflow/include/great_expectations/great_expectations.yml`. Could be a problem if the DAG got moved up a level in the hierarchy, it should be under `dags/great_expectations/ge_dag.py`

Also I think the content in this section is wrong, I'm not sure what

> the operator detects your Astro OpenLineage configuration from the `include` folder automatically as long as you set

is supposed to imply. The operator only detects an OpenLineage environment, irrespective of Astro, if you are using the operator without supplying a checkpoint, when it builds its default action list. It has nothing to do with the data context.

File: astro/set-up-data-lineage.md [hunk 3]
```diff
@@ -160,15 +160,41 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+   )
+   # Set base path for Data Context 
+   base_path = Path(__file__).parents[2]
+
+   ...
+
+   # Example task using GreatExpectationsOperator 
+   ge_task = GreatExpectationsOperator(
+    task_id="ge_task",
+    # Set directory fot the Data Context
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:188]: nit: fot --> for

File: astro/set-up-data-lineage.md [hunk 4]
```diff
@@ -160,15 +160,41 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
-- A [Great Expectations project](https://legacy.docs.greatexpectations.io/en/latest/guides/tutorials/getting_started.html#tutorials-getting-started).
-- Your Astro base domain.
-- Your Organization's OpenLineage API key.
+- A [Great Expectations Data Context](https://legacy.docs.greatexpectations.io/en/latest/guides/tutorials/getting_started.html#tutorials-getting-started)
+- If using a custom Checkpoint or Checkpoint config, your Astro base domain and OpenLineage API key.
 
 #### Setup
 
-If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use a custom Checkpoint or Checkpoint Config, the operator detects your Astro OpenLineage configuration and sends lineage information automatically. If you use custom Checkpoints, complete the following steps:
+1. Make your Data Context accessible to your DAGs. For most use cases, Astronomer recommends adding the Data Context to your Astro project `include` folder, then adding the following lines to a given DAG:
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:168]: I'd reword this to "recommends adding your Great Expectations project to your Astro project `include` folder. The GreatExpectationsOperator will access `include/great_expectations/great_expectations.yml` and use the configuration to run your Expectations. Then add the following lines to a given DAG:"

File: astro/set-up-data-lineage.md [hunk 5]
```diff
@@ -160,15 +160,41 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+    ge_root_dir=os.path.join(base_path, "include", "great_expectations"),
+   )
+   ```
 
-1. Update your `great_expectations.yml` file to add `OpenLineageValidationAction` to your `action_list_operator` configuration:
+   The GreatExpectationsOperator will access `include/great_expectations/great_expectations.yml` and use the configuration to run your Expectations.
+
+   If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use a custom Checkpoint or Checkpoint Config, you can skip steps 2 and 3.
+   
+2. Update your `great_expectations.yml` file to add `OpenLineageValidationAction` to your `action_list_operator` configuration:
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:197]: None of lines 197-211 are accurate; the checkpoint file is what needs to be updated and an `OpenLineageValidationAction` added to the `action_list`, like so:

```yml
name: my.simple.chk
config_version: 1.0
template_name:
module_name: great_expectations.checkpoint
class_name: Checkpoint
run_name_template:
expectation_suite_name:
batch_request: {}
action_list:
  - name: store_validation_result
    action:
      class_name: StoreValidationResultAction
  - name: store_evaluation_params
    action:
      class_name: StoreEvaluationParametersAction
  - name: update_data_docs
    action:
      class_name: UpdateDataDocsAction
  - name: open_lineage
    action:
      class_name: OpenLineageValidationAction,
      module_name: openlineage.common.provider.great_expectations,
      openlineage_host: https://<your-astro-base-domain>,
      openlineage_apiKey: <openlineage_api_key>,
      openlineage_namespace: <NAMESPACE_NAME> # Replace with your job namespace; Astronomer recommends using a meaningful namespace such as `dev` or `prod`,
      job_name: validate_task_name,
```

File: astro/set-up-data-lineage.md [hunk 6]
```diff
@@ -179,15 +205,15 @@ If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use
             action:
               class_name: OpenLineageValidationAction
               module_name: openlineage.common.provider.great_expectations
-              openlineage_host: https://<your-astro-base-domain>
+              openlineage_host: https://astro-<your-base-domain>.datakin.com
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:208]: I'm pretty sure this is still just `https://<your-astro-base-domain>`, we're not doing the redirects to Datakin anymore. @rossturk can confirm

File: astro/set-up-data-lineage.md [hunk 7]
```diff
@@ -179,15 +205,15 @@ If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use
...
               openlineage_apiKey: <your-lineage-api-key>
               openlineage_namespace: <NAMESPACE_NAME> # Replace with your job namespace; Astronomer recommends using a meaningful namespace such as `dev` or `prod`.
               job_name: validate_my_dataset
     ```
 
-2. Lineage support for GreatExpectations requires the use of the `ActionListValidationOperator`. In each of your checkpoint's xml files in `checkpoints/`, set the `validation_operator_name` configuration to `action_list_operator`:
+3. In each of your Checkpoint's YAML files in `checkpoints/`, set the `validation_operator_name` configuration to `action_list_operator` like in the following snippet:
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:214]: This step can be removed, it is incorrect and not needed. The _only_ configuration for lineage data emission is adding the action to the action list in the checkpoint.

**Review by denimalpaca — CHANGES_REQUESTED** [review]

File: astro/set-up-data-lineage.md [hunk 8]
```diff
@@ -160,15 +160,41 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+   # Example task using GreatExpectationsOperator 
+   ge_task = GreatExpectationsOperator(
+    task_id="ge_task",
+    # Set directory fot the Data Context
+    ge_root_dir=os.path.join(base_path, "include", "great_expectations"),
+   )
+   ```
 
-1. Update your `great_expectations.yml` file to add `OpenLineageValidationAction` to your `action_list_operator` configuration:
+   The GreatExpectationsOperator will access `include/great_expectations/great_expectations.yml` and use the configuration to run your Expectations.
```
Comment (jwitz) [inline astro/set-up-data-lineage.md:193]:
```suggestion
```

File: astro/set-up-data-lineage.md (hunk 3 above)
Comment (jwitz) [inline astro/set-up-data-lineage.md:188]:
```suggestion
    # Set directory for the Data Context
```

File: astro/set-up-data-lineage.md (hunk 4 above)
Comment (jwitz) [inline astro/set-up-data-lineage.md:168]:
```suggestion
1. Make your Data Context accessible to your DAGs. For most use cases, Astronomer recommends adding the Data Context to your Astro project `include` folder. The GreatExpectationsOperator will access `include/great_expectations/great_expectations.yml` and use the configuration to run your Expectations. Then, add the following lines to your DAGs: 
```

File: astro/set-up-data-lineage.md [hunk 9]
```diff
@@ -160,42 +160,61 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+    ```python
+    # Required imports for Great Expectations
+    import os
+    from pathlib import Path
+    from great_expectations_provider.operators.great_expectations import GreatExpectationsOperator
+    from great_expectations.core.batch import BatchRequest
+    from great_expectations.data_context.types.base import (
+      DataContextConfig,
+      CheckpointConfig
+   )
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:179]: These are *not* required imports. `BatchRequest` probably should not be imported in most cases, and `DataContextConfig` and `CheckpointConfig` are both the in-memory versions of the Data Context and Checkpoint, and neither are used here.

File: astro/set-up-data-lineage.md [hunk 10]
```diff
@@ -160,42 +160,61 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+   base_path = Path(__file__).parents[2]
+
+   ...
+
+   # Example task using GreatExpectationsOperator 
+   ge_task = GreatExpectationsOperator(
+    task_id="ge_task",
+    # Set directory for the Data Context
+    ge_root_dir=os.path.join(base_path, "include", "great_expectations"),
+   )
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:190]: I don't like showing the operator like this because the DAG wouldn't even import. Maybe we can add some ellipsis to imply other parameters are required?

```
ge_task = GreatExpectationsOperator(
    task_id="ge_task",
    # Set directory for the Data Context
    ge_root_dir=os.path.join(base_path, "include", "great_expectations"),
    ...
   )
```

File: astro/set-up-data-lineage.md [hunk 11]
```diff
@@ -160,42 +160,61 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+   ```
 
-    ```xml
-    name: check_users
-    config_version:
+   If you use the `GreatExpectationsOperator` version 0.2.0 or later and don't use a custom Checkpoint or Checkpoint Config, you can skip steps 2 and 3.
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:193]: I think "custom Checkpoint" should be reworded to "Checkpoint file" as "custom" implies that a "regular" Checkpoint being used means to also skip steps 2 and 3.

File: astro/set-up-data-lineage.md [hunk 12]
```diff
@@ -160,42 +160,61 @@ This guide outlines how to set up lineage collection for a Great Expectations pr
...
+    class_name: Checkpoint
+    run_name_template:
+    expectation_suite_name:
+    batch_request: {}
+    action_list:
+      - name: open_lineage
+        action:
+          class_name: OpenLineageValidationAction,
+          module_name: openlineage.common.provider.great_expectations,
+          openlineage_host: https://astro-<your-astro-base-domain>.datakin.com,
```
Comment (denimalpaca) [inline astro/set-up-data-lineage.md:211]: I would like to verify this line with someone from the OL team. @conorbev @rossturk

**Review by denimalpaca — CHANGES_REQUESTED** [review]
Getting closer ;)

**Review by denimalpaca — APPROVED** [review]
lgtm!
